libraries/aws_hosted_zone.rb

```diff
@@ -0,0 +1,96 @@
+require 'aws_backend'
+
+class AwsHostedZone < AwsResourceBase
+  name 'aws_hosted_zone'
+  desc 'Verifies hosted zone settings are correct'
+
+  example "
+    describe aws_hosted_zone('zone-name') do
+      it { should exist }
+      its ('name_servers.count') { should eq 4 }
+      its ('private_zone') { should be false }
+      its ('record_names') { should include 'sid-james.carry-on.films.com' }
+    end
+  "
+
+  attr_reader :name_servers, :private_zone, :record_count, :records
+
+  FilterTable.create
+             .register_column(:record_names,      field: :record_name)
+             .register_column(:ids,               field: :id)
+             .register_column(:private,           field: :private)
+             .register_column(:ttls,              field: :ttl)
+             .register_column(:types,             field: :type)
+             .register_column(:values,            field: :value)
+             .install_filter_methods_on_resource(self, :records)
+
+  def initialize(opts = {})
+    opts = { zone_name: opts } if opts.is_a?(String)
+    super(opts)
+
+    validate_parameters(required: [:zone_name])
+
+    get_zone_id(opts[:zone_name])
+
+    get_zone_details(@id) if !@id.nil?
```

Comment:
this could be simplified to `unless exist?`

Reply:
I did originally, but the linter complained " Use a guard clause instead of wrapping the code inside a conditional expression".  All other instances fixed.